test/builders/remote/file.js

```diff
@@ -11,17 +13,32 @@ export default class RemoteFileBuilder extends RemoteBaseBuilder {
   constructor (cozy) {
     super(cozy)
 
-    this.data = `Content of remote file ${fileNumber}`
+    this._data = `Content of remote file ${fileNumber}`
 
     Object.assign(this.options, {
       name: `remote-file-${fileNumber++}`,
       contentType: undefined
     })
   }
 
+  contentType (contentType) {
+    this.options.contentType = contentType
+    return this
+  }
+
+  data (data) {
+    this._data = data
+    return this
+  }
+
+  dataFromFile (path) {
+    this._data = fs.readFileSync(path)
```

Comment:
`fs.readFileSync()` [returns a `Buffer`](https://nodejs.org/docs/latest-v4.x/api/fs.html#fs_fs_readfilesync_file_options) since no encoding argument is given.

Node's `Buffer` class [implements the Uint8Array API](https://nodejs.org/docs/latest-v4.x/api/buffer.html#buffer_buffer).

```
> let b = fs.readFileSync('./test/fixtures/cool-pillow.jpg')
> b
<Buffer ...>
> b instanceof Buffer
true
> b instanceof Uint8Array
true
> b.buffer
ArrayBuffer { byteLength: 21102 }
> b.buffer instanceof ArrayBuffer
true
```

But upload fails as if the `ArrayBuffer` was converted to `String`:

```
> b.buffer.toString()
'[object ArrayBuffer]'
```

```
$ cat storage/test.cozy-desktop.local\:8080/pillow.jpg 
[object ArrayBuffer]
```